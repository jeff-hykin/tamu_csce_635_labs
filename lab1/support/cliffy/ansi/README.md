# Ansi

<p align="center" class="badges-container">
  <a href="https://github.com/c4spar/deno-cliffy/releases">
    <img alt="Version" src="https://img.shields.io/github/v/release/c4spar/deno-cliffy?logo=github&color=blue" />
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/actions/workflows/test.yml">
    <img alt="Build status" src="https://github.com/c4spar/deno-cliffy/workflows/Test/badge.svg?branch=main" />
  </a>
  <a href="https://codecov.io/gh/c4spar/deno-cliffy">
    <img src="https://codecov.io/gh/c4spar/deno-cliffy/branch/main/graph/badge.svg"/>
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/labels/module%3Aansi-escape">
    <img alt="issues" src="https://img.shields.io/github/issues/c4spar/deno-cliffy/module:ansi-escape?label=issues&logo=github&color=yellow">
  </a>
  <a href="https://deno.land/">
    <img alt="Deno version" src="https://img.shields.io/badge/deno-^1.4.0-blue?logo=deno" />
  </a>
  <a href="https://doc.deno.land/https/deno.land/x/cliffy/ansi/mod.ts">
    <img alt="doc" src="https://img.shields.io/badge/deno-doc-yellow?logo=deno" />
  </a>
  <a href="https://discord.gg/ghFYyP53jb">
    <img alt="Discord" src="https://img.shields.io/badge/join-chat-blue?logo=discord&logoColor=white" />
  </a>
  <a href="../LICENSE">
    <img alt="License" src="https://img.shields.io/github/license/c4spar/deno-cliffy?logo=github" />
  </a>
  <br>
  <a href="https://deno.land/x/cliffy">
    <img alt="Discord" src="https://img.shields.io/badge/Published on deno.land-blue?logo=deno&logoColor=959DA6&color=272727" />
  </a>
</p>

The documentation has been moved to https://cliffy.io/docs/ansi.
