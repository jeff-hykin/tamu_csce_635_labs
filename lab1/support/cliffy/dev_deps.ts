/* std */
export {
  assert,
  assertEquals,
  assertInstanceOf,
  assertRejects,
  assertStrictEquals,
  assertThrows,
} from "https://deno.land/std@0.196.0/testing/asserts.ts";
export {
  assertSpyCall,
  assertSpyCalls,
  spy,
} from "https://deno.land/std@0.196.0/testing/mock.ts";
export { assertSnapshot } from "https://deno.land/std@0.196.0/testing/snapshot.ts";
export { describe, it } from "https://deno.land/std@0.196.0/testing/bdd.ts";
export {
  assertType,
  type IsExact,
} from "https://deno.land/std@0.196.0/testing/types.ts";
export {
  bold,
  red,
  stripColor,
} from "https://deno.land/std@0.196.0/fmt/colors.ts";
export { dirname } from "https://deno.land/std@0.196.0/path/mod.ts";
export { expandGlob } from "https://deno.land/std@0.196.0/fs/expand_glob.ts";
export type { WalkEntry } from "https://deno.land/std@0.196.0/fs/walk.ts";
export { copy } from "https://deno.land/std@0.196.0/streams/copy.ts";
export { format } from "https://deno.land/std@0.196.0/datetime/format.ts";
export { gt } from "https://deno.land/std@0.196.0/semver/gt.ts";
export { lt } from "https://deno.land/std@0.196.0/semver/lt.ts";

/* 3rd party */
export { default as sinon } from "https://cdn.skypack.dev/sinon@v13.0.2?dts";
