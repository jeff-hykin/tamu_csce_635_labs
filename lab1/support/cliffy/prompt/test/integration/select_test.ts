import { ansi } from "../../../ansi/ansi.ts";
import { format } from "../../../dev_deps.ts";
import { Select } from "../../select.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";

await snapshotTest({
  name: "select prompt > should select an option",
  meta: import.meta,
  osSuffix: ["windows"],
  stdin: ansi
    .cursorDown
    .cursorDown
    .text("\n")
    .toArray(),
  async fn() {
    await Select.prompt({
      message: "Select an option",
      options: [
        { name: "Foo", value: "foo" },
        { name: "Bar", value: "bar" },
        { name: "Baz", value: "baz" },
      ],
    });
  },
});

await snapshotTest({
  name: "select prompt > should search an option",
  meta: import.meta,
  osSuffix: ["windows"],
  stdin: ansi
    .text("baz")
    .text("\n")
    .toArray(),
  async fn() {
    await Select.prompt({
      message: "Select an option",
      search: true,
      options: [
        { name: "Foo", value: "foo" },
        { name: "Bar", value: "bar" },
        { name: "Baz", value: "baz" },
      ],
    });
  },
});

await snapshotTest({
  name: "select prompt > should format option value",
  meta: import.meta,
  osSuffix: ["windows"],
  stdin: ansi
    .cursorDown
    .text("\n")
    .toArray(),
  async fn() {
    await Select.prompt({
      message: "Message...",
      options: [
        { value: new Date(10000) },
        { value: new Date(20000) },
      ],
      format: (date) => format(date, "dd-MM-yyyy"),
    });
  },
});

await snapshotTest({
  name: "select prompt > should support separator option",
  meta: import.meta,
  osSuffix: ["windows"],
  stdin: ansi
    .cursorDown
    .text("\n")
    .toArray(),
  async fn() {
    await Select.prompt({
      message: "Message...",
      options: [
        { value: new Date(10000) },
        Select.separator("+++++"),
        { value: new Date(20000) },
      ],
      format: (date) => format(date, "dd-MM-yyyy"),
    });
  },
});

await snapshotTest({
  name: "select prompt > should not select disabled option",
  meta: import.meta,
  osSuffix: ["windows"],
  stdin: ansi
    .cursorForward
    .cursorBackward
    .cursorBackward
    .cursorBackward
    .text("\n")
    .toArray(),
  async fn() {
    await Select.prompt({
      message: "Pick a value",
      options: [
        { name: "Value1", value: "value-1", disabled: true },
        { name: "Value2", value: "value-2" },
        { name: "Value3", value: "value-3" },
        { name: "Value4", value: "value-4", disabled: true },
        { name: "Value5", value: "value-5" },
        { name: "Value6", value: "value-6" },
        { name: "Value7", value: "value-7" },
        { name: "Value8", value: "value-8", disabled: true },
        { name: "Value9", value: "value-9", disabled: true },
        { name: "Value10", value: "value-10", disabled: true },
        { name: "Value11", value: "value-11", disabled: true },
        { name: "Value12", value: "value-12", disabled: true },
        { name: "Value13", value: "value-13" },
        { name: "Value14", value: "value-14" },
        { name: "Value15", value: "value-15" },
        { name: "Value16", value: "value-16" },
        { name: "Value17", value: "value-17" },
        { name: "Value18", value: "value-18" },
        { name: "Value19", value: "value-19" },
      ],
    });
  },
});
