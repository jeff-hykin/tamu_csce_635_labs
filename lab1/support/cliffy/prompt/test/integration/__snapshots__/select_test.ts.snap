export const snapshot = {};

snapshot[`select prompt > should select an option 1`] = `
stdout:
"? Select an option
‚ùØ Foo
  Bar
  Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  Foo
‚ùØ Bar
  Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  Foo
  Bar
‚ùØ Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option ‚Ä∫ Baz
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`select prompt > should search an option 1`] = `
stdout:
"? Select an option üîé 
‚ùØ Foo
  Bar
  Baz\\x1b[3A\\x1b[23G\\x1b[G\\x1b[0J
? Select an option üîé baz
‚ùØ Baz\\x1b[1A\\x1b[26G\\x1b[G\\x1b[0J
? Select an option ‚Ä∫ Baz
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`select prompt > should format option value 1`] = `
stdout:
"? Message...
‚ùØ 01-01-1970
  01-01-1970\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Message...
  01-01-1970
‚ùØ 01-01-1970\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Message... ‚Ä∫ 01-01-1970
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`select prompt > should support separator option 1`] = `
stdout:
"? Message...
‚ùØ 01-01-1970
  +++++
  01-01-1970\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Message...
  01-01-1970
  +++++
‚ùØ 01-01-1970\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Message... ‚Ä∫ 01-01-1970
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`select prompt > should not select disabled option 1`] = `
stdout:
"? Pick a value
  Value1
‚ùØ Value2
  Value3
  Value4
  Value5
  Value6
  Value7
  Value8
  Value9
  Value10\\x1b[10A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Pick a value
  Value10
  Value11
  Value12
‚ùØ Value13
  Value14
  Value15
  Value16
  Value17
  Value18
  Value19\\x1b[10A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Pick a value
  Value1
  Value2
‚ùØ Value3
  Value4
  Value5
  Value6
  Value7
  Value8
  Value9
  Value10\\x1b[10A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Pick a value
  Value1
‚ùØ Value2
  Value3
  Value4
  Value5
  Value6
  Value7
  Value8
  Value9
  Value10\\x1b[10A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Pick a value
  Value1
‚ùØ Value2
  Value3
  Value4
  Value5
  Value6
  Value7
  Value8
  Value9
  Value10\\x1b[10A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Pick a value ‚Ä∫ Value2
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
